x/sequencer/keeper/sequencer.go

```diff
@@ -1,28 +1,56 @@
 package keeper
 
 import (
+	"time"
+
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/dymensionxyz/dymension/v3/x/sequencer/types"
 )
 
 // SetSequencer set a specific sequencer in the store from its index
 func (k Keeper) SetSequencer(ctx sdk.Context, sequencer types.Sequencer) {
-	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.SequencerKeyPrefix))
+	store := ctx.KVStore(k.storeKey)
 	b := k.cdc.MustMarshal(&sequencer)
 	store.Set(types.SequencerKey(
 		sequencer.SequencerAddress,
 	), b)
+
+	seqByrollappKey := types.SequencerByRollappByStatusKey(sequencer.RollappId, sequencer.SequencerAddress, sequencer.Status)
+	store.Set(seqByrollappKey, b)
+
+	//To support InitGenesis scenario
+	if sequencer.Status == types.Unbonding {
+		k.setUnbondingSequencerQueue(ctx, sequencer)
+	}
 }
 
-// GetSequencer returns a sequencer from its index
-func (k Keeper) GetSequencer(
-	ctx sdk.Context,
-	sequencerAddress string,
+// Update sequencer status
+func (k Keeper) UpdateSequencer(ctx sdk.Context, sequencer types.Sequencer, oldStatus types.OperatingStatus) {
+	store := ctx.KVStore(k.storeKey)
+	b := k.cdc.MustMarshal(&sequencer)
+	store.Set(types.SequencerKey(sequencer.SequencerAddress), b)
 
-) (val types.Sequencer, found bool) {
-	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.SequencerKeyPrefix))
+	oldKey := types.SequencerByRollappByStatusKey(sequencer.RollappId, sequencer.SequencerAddress, oldStatus)
+	store.Delete(oldKey)
 
+	seqByrollappKey := types.SequencerByRollappByStatusKey(sequencer.RollappId, sequencer.SequencerAddress, sequencer.Status)
+	store.Set(seqByrollappKey, b)
+
+	//handle unbonding queue changes due to status change
+	if sequencer.Status == oldStatus {
+		return
+	}
+	if sequencer.Status == types.Unbonding {
+		k.setUnbondingSequencerQueue(ctx, sequencer)
+	} else if oldStatus == types.Unbonding {
```

Comment:
I guess the assumption here is always that from unbonding we can only go to unbonded. maybe would be cleaner to check if `sequencer.Status == types.Unbonded` in case we'll later on add the ability to go from unbonding back to bonded.

Reply:
actually, there's no such assumption.
I assume that if it was unbonding, and now it's not, we should remove it from the unbonding queue